app/validators/seat_availability_validator.rb

```diff
@@ -0,0 +1,9 @@
+class SeatAvailabilityValidator < ActiveModel::Validator
+  def validate(record)
+    status = record.seance.seats_hash[record.seat]
+    seat = record.seat
```

Comment:
It is good practice to declare variables before uses if possible. If status will be `reserved` this `seat`  variable wont do anything.
You can move it behind return or use `record.seat `directly in error.message